### PR TITLE
2sideband modulation option and settings dict for MultiSine

### PR DESCRIPTION
**Pull request by RostyslavSavytskyy**
Added an option for modulating only with I or Q quadrature which gives us 2 sideband modulation. Also modified MultiSinePulseImplementation to the new format with the settings dictionary.

### CODE REVIEW

File: silq/instrument_interfaces/keysight/E8267D_interface.py [hunk 1]
```diff
@@ -671,7 +695,7 @@ def get_additional_pulses(self, interface: InstrumentInterface):
                             'input_instrument': interface.instrument_name(),
                             'input_channel': 'trig_in'})]
 
-        assert (interface.IQ_modulation() == 'on') and \
+        assert (IQ_modulation == 'on') and \
                (interface.FM_mode() == 'IQ'), 'FM_mode should be IQ and IQ_modulation should be ON ' \
```
Comment (maij) [inline silq/instrument_interfaces/keysight/E8267D_interface.py:699]: Is FM_mode part of `frequency_settings`? Or can you only access this via the interface?
I'm a bit out of the loop, I'm not sure when the interface was changed to pass more variables than just the interface...

**Review by maij — COMMENTED** [review]
Looks good, I just have 1 question about how certain variables are passed to the pulse implementation functions.

File: silq/instrument_interfaces/keysight/E8267D_interface.py (hunk 1 above)
Comment (RostyslavSavytskyy) [reply silq/instrument_interfaces/keysight/E8267D_interface.py:699]: No, so Serwan decided to detach some of the settings from the interface to make functions more independent, but not everything is in there, like FM_mode, etc. However, the interface still contains all of the parameters.

Comment (maij) [reply silq/instrument_interfaces/keysight/E8267D_interface.py:699]: Righto, I guess that's something that should be consolidated in future.

**Review by maij — APPROVED** [review]

**Pull request merged**